aws-sdk-resources/lib/aws-sdk-resources/services/s3/object_copier.rb

```diff
@@ -12,6 +12,7 @@ def initialize(object, options = {})
       end
 
       def copy_from(source, options = {})
+        options[:copy_source_client] = source.client if source.is_a?(S3::Object)
```

Comment:
Sounds good- I'll try to collect the feedback and push an update this afternoon.